explorer/interpreter/interpreter.cpp

```diff
@@ -615,7 +622,8 @@ auto Interpreter::InstantiateType(Nonnull<const Value*> type,
       CARBON_ASSIGN_OR_RETURN(
           Nonnull<const Bindings*> bindings,
           InstantiateBindings(&class_type.bindings(), source_loc));
-      return arena_->New<NominalClassType>(&class_type.declaration(), bindings);
+      return arena_->New<NominalClassType>(&class_type.declaration(), bindings,
+                                           class_type.base());
```

Comment:
Is the `base` value here intended to have already been substituted? Eg, given:
```
class A(T:! Type) {}
class B(T:! Type) extends A(T) {}
fn F[T:! Type](b: B(T)) {}
```
... in a call to `F` with `T = i32`, is the type of `B` supposed to have a base of `A(T)` or a base of `A(i32)`? If the latter, then you'll need to call `InstantiateType` here on the base to replace the `T` with `i32` during evaluation.

Reply:
The base should be subsituted as well yes. Thanks!

Reply:
Runnning a basic parametrized base class example fails:
```
base class A(T:! Type) {}
// CHECK:STDERR: COMPILATION ERROR: {{.*}}/explorer/testdata/class/fail_parametrized_base_class.carbon:[[@LINE+1]]: could not resolve 'T'
class B(T:! Type) extends A(T) {}
fn F[T:! Type](b: B(T)) {}
```
Digging a bit, the resolution error occurs while checking an `ImplDeclaration`, specifically `enclosing_scope.Resolve("Self", impl.source_loc()).ok()`. I started looking in the parser to see if the class declaration had an issue, but unsure yet. Still pretty unfamiliar with how generics in general are handled, so I could be missing something.